KeyVaultRecoverySamples.cs

```diff
@@ -70,39 +101,44 @@ public static async Task DemonstrateRecoveryAndPurgeForNewVaultAsync()
                 Console.WriteLine("done.");
 
                 Console.Write("Retrieving newly created vault...");
-                var retrievedVault = await sample.ManagementClient.Vaults.GetAsync(rgName, vaultName).ConfigureAwait(false);
+                var retrievedVault = (await rg.GetKeyVaultAsync(vaultName)).Value;
                 Console.WriteLine("done.");
 
                 // delete vault
                 Console.Write("Deleting vault...");
-                await sample.ManagementClient.Vaults.DeleteAsync(rgName, vaultName).ConfigureAwait(false);
+                await retrievedVault.DeleteAsync(WaitUntil.Completed);
                 Console.WriteLine("done.");
 
                 // confirm the existence of the deleted vault
                 Console.Write("Retrieving deleted vault...");
-                deletedVault = await sample.ManagementClient.Vaults.GetDeletedAsync(vaultName, retrievedVault.Location).ConfigureAwait(false);
-                Console.WriteLine("done; '{0}' deleted on: {1}, scheduled for purge on: {2}", deletedVault.Id, deletedVault.Properties.DeletionDate, deletedVault.Properties.ScheduledPurgeDate);
+                ResourceIdentifier rid = DeletedKeyVaultResource.CreateResourceIdentifier(sub.Data.SubscriptionId, location, vaultName);
+                DeletedKeyVaultResource deletedRes = await client.GetDeletedKeyVaultResource(rid).GetAsync();
+                Console.WriteLine("done; '{0}' deleted on: {1}, scheduled for purge on: {2}", deletedRes.Data.Id, deletedRes.Data.Properties.DeletedOn, deletedRes.Data.Properties.ScheduledPurgeOn);
 
                 // recover; set the creation mode as 'recovery' in the vault parameters
                 Console.Write("Recovering deleted vault...");
-                vaultParameters.Properties.CreateMode = CreateMode.Recover;
-                await sample.ManagementClient.Vaults.CreateOrUpdateAsync(rgName, vaultName, vaultParameters).ConfigureAwait(false);
+                var keyVaultSku_recover = new KeyVaultSku(KeyVaultSkuFamily.A, KeyVaultSkuName.Standard);
```

Comment:
Is a new key vault being provisioned here?

Reply:
well, there is no need to create a new KeyVaultSku since I used the varialbe 'keyVaultSku' declared above  in the next line.
I'll fix this right now.